GitUI/Theming/ThemeRepository.cs

```diff
@@ -86,19 +89,58 @@ private IEnumerable<ThemeId> GetUserCustomizedThemeIds()
                 : Enumerable.Empty<ThemeId>();
         }
 
+        string ICssUrlResolver.ResolveCssUrl(string url)
+        {
+            if (url.EndsWith(Extension, StringComparison.OrdinalIgnoreCase))
+            {
+                url = url.Substring(0, url.Length - Extension.Length);
+            }
+
+            ThemeId id = url.StartsWith(CssVariableUserThemesDirectory)
+                ? new ThemeId(url.Substring(CssVariableUserThemesDirectory.Length), isBuiltin: false)
+                : new ThemeId(url, isBuiltin: true);
+
+            try
+            {
+                return GetPath(id);
+            }
+            catch (ThemeNotFoundException)
+            {
+                return null;
+            }
+        }
+
         private string GetPath(ThemeId id)
         {
+            string path;
             if (id.IsBuiltin)
             {
-                return Path.Combine(AppThemesDirectory, id.Name + Extension);
+                path = Path.Combine(AppThemesDirectory, id.Name + Extension);
             }
+            else
+            {
+                if (UserThemesDirectory == null)
+                {
+                    throw new ThemeNotFoundException("There is no directory for custom user themes in portable mode");
+                }
 
-            if (UserThemesDirectory == null)
+                path = Path.Combine(UserThemesDirectory, id.Name + Extension);
+            }
+
+            if (!File.Exists(path))
             {
-                throw new InvalidOperationException("There is no directory for custom user themes in portable mode");
+                throw new ThemeNotFoundException($"Theme file not found: {path}");
             }
 
-            return Path.Combine(UserThemesDirectory, id.Name + Extension);
+            return path;
+        }
+
+        private class ThemeNotFoundException : Exception
```

Comment:
```suggestion
        private class ThemeNotFoundException : FileNotFoundException
```
https://docs.microsoft.com/dotnet/api/system.io.filenotfoundexception.-ctor#System_IO_FileNotFoundException__ctor_System_String_System_String_

Reply:
kept 2 specialized exceptions

- `CssUrlResolverException`
- `ThemeRepositoryException` any "expected" exception when loading theme, wrapping any io, security and or `CssUrlResolverException`